src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int MAGIC_NUMBER = 6;
+
+    public static void main(String[] args) {
+        Figure[] figuresArray = new Figure[MAGIC_NUMBER];
+        FigureSupplier supplier = new FigureSupplier();
+
+        for (int i = 0; i < 2; i++) {

```

Comment:
You need to iterate through the entire array, but with an additional check (if this first half of array) inside the loop.